packages/replay_cubit/lib/src/replay_cubit.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:cubit/cubit.dart';
+
+import 'change_stack.dart';
+
+/// {@template hydrated_cubit}
+/// Specialized [Cubit] which handles initializing the [Cubit] state
+/// based on the persisted state. This allows state to be persisted
+/// across application restarts.
+/// {@endtemplate}
+abstract class ReplayCubit<State> extends Cubit<State> {
+  /// {@macro hydrated_cubit}
+  ReplayCubit(State state, {int maxChanges}) : super(state) {
+    _changeStack = ChangeStack<State>(max: maxChanges);
+  }
+
+  /// Instance of [Storage] which will be used to
+  /// manage persisting/restoring the [Cubit] state.
+  static ChangeStack _changeStack;
```

Comment:
Does the `ChangeStack` instance need to be `static`?

```suggestion
  ReplayCubit(State state, {int maxChanges}) : _changeStack = ChangeStack<State>(max: maxChanges), super(state);

  /// Instance of [Storage] which will be used to
  /// manage persisting/restoring the [Cubit] state.
  final ChangeStack _changeStack;
```

Reply:
not unless it is created multiple times. On hydrated bloc storage is static